feat: SmartEscrow fee voting changes

mvadari opened the pull request:
## High Level Overview of Change

<!--
Please include a summary of the changes.
This may be a direct input to the release notes.
If too broad, please consider splitting into multiple PRs.
If a relevant task or issue, please link it here.
-->

### Context of Change

Stack of PRs (from `develop` outwards):
* https://github.com/XRPLF/rippled/pull/5999
* https://github.com/XRPLF/rippled/pull/6075
* This PR
* https://github.com/XRPLF/rippled/pull/6329
* https://github.com/XRPLF/rippled/pull/5811

### Type of Change

<!--
Please check [x] relevant options, delete irrelevant ones.
-->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Performance (increase or change in throughput and/or latency)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [ ] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

### API Impact

<!--
Please check [x] relevant options, delete irrelevant ones.

* If there is any impact to the public API methods (HTTP / WebSocket), please update https://github.com/xrplf/rippled/blob/develop/API-CHANGELOG.md
  * Update API-CHANGELOG.md and add the change directly in this PR by pushing to your PR branch.
* libxrpl: See https://github.com/XRPLF/rippled/blob/develop/docs/build/depend.md
* Peer Protocol: See https://xrpl.org/peer-protocol.html
-->

- [ ] Public API: New feature (new methods and/or new fields)
- [ ] Public API: Breaking change (in general, breaking changes should only impact the next api_version)
- [ ] `libxrpl` change (any change that may affect `libxrpl` or dependents of `libxrpl`)
- [ ] Peer protocol change (must be backward compatible or bump the peer protocol version)

<!--
## Before / After
If relevant, use this section for an English description of the change at a technical level.
If this change affects an API, examples should be included here.

For performance-impacting changes, please provide these details:
1. Is this a new feature, bug fix, or improvement to existing functionality?
2. What behavior/functionality does the change impact?
3. In what processing can the impact be measured? Be as specific as possible - e.g. RPC client call, payment transaction that involves LOB, AMM, caching, DB operations, etc.
4. Does this change affect concurrent processing - e.g. does it involve acquiring locks, multi-threaded processing, or async processing?
-->

<!--
## Test Plan
If helpful, please describe the tests that you ran to verify your changes and provide instructions so that others can reproduce.
This section may not be needed if your change includes thoroughly commented unit tests.
-->

<!--
## Future Tasks
For future tasks related to PR.
-->

mvadari committed:
SmartEscrow fee voting changes

mvadari committed:
Merge branch 'ripple/wamr-host-functions' into ripple/se/fees

mvadari committed:
fix issues

mvadari committed:
Merge branch 'ripple/wamr-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wamr-host-functions' into ripple/se/fees

mvadari committed:
fix tests

mvadari committed:
Merge branch 'ripple/wamr-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wamr-host-functions' into ripple/se/fees

mvadari committed:
test: add tests for fee voting (#5747)

mvadari committed:
Merge branch 'ripple/wamr-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wamr-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wamr-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wamr-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wamr-host-functions' into ripple/se/fees

mvadari committed:
fix ledger used for rules

mvadari committed:
Merge branch 'ripple/wamr-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'wamr-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'wamr-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'wamr-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'wamr-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wamr-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge remote-tracking branch 'upstream/ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge remote-tracking branch 'upstream/ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
fix more merge issues

mvadari committed:
Merge branch 'wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'wasmi-host-functions' into ripple/se/fees

mvadari committed:
simplify fee code (#6249)

* simplify lambda

* clean up fee code

* fix tests, better error handling

* simplify source_location

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge commit '4eb34f3' into ripple/se/fees

mvadari committed:
Merge remote-tracking branch 'upstream/ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
fix merge issues

mvadari committed:
Merge branch 'wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge remote-tracking branch 'upstream/ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'wasmi-host-functions' into ripple/se/fees

mvadari committed:
fix comments

mvadari committed:
Merge branch 'wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' of https://github.com/XRPLF/rippled into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' of https://github.com/XRPLF/rippled into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' of https://github.com/XRPLF/rippled into ripple/se/fees

mvadari committed:
fix clang-tidy issues

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' of https://github.com/XRPLF/rippled into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' of https://github.com/XRPLF/rippled into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' of https://github.com/XRPLF/rippled into ripple/se/fees

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' into ripple/se/fees

mvadari committed:
Update cfg/xrpld-example.cfg

Co-authored-by: xrplf-ai-reviewer[bot] <266832837+xrplf-ai-reviewer[bot]@users.noreply.github.com>

mvadari committed:
Update src/test/app/FeeVote_test.cpp

Co-authored-by: xrplf-ai-reviewer[bot] <266832837+xrplf-ai-reviewer[bot]@users.noreply.github.com>

mvadari committed:
Merge branch 'ripple/wasmi-host-functions' into ripple/se/fees-fix-tidy

mvadari committed:
fix more build issues

mvadari committed:
fix clang-tidy

mvadari committed:
fix issues

mvadari committed:
Merge pull request #7301 from XRPLF/ripple/se/fees-fix-tidy

chore: Update branch from `develop`

pwang200 commented on src/xrpld/app/misc/FeeVoteImpl.cpp:
since this is a warn level log, probably worth adding the new fields, if amendment is active

mvadari committed:
feat: Add fee vote maxes (#7346)

This PR adds hard-cap maxes for the size and compute limits, currently set at twice the defaults.

pwang200 reviewed (approved):
I have a suggestion about logging, otherwise looks good.

mvadari committed:
Merge branch 'ripple/se/wasmi-tests' into ripple/se/fees

TimothyBanks commented on src/libxrpl/ledger/Ledger.cpp:
```diff
@@ -598,6 +606,22 @@ Ledger::setup()
+                        dest = src.value();
+                    }
+                };
+                assign(fees_.extensionComputeLimit, extensionComputeLimit);
```
How about something like this?
```suggestion
                assign(fees_.extensionComputeLimit, extensionComputeLimit);
                fees_.extensionComputeLimit = extensionComputeLimit.value_or(fees_.extensionComputeLimit)
```
or in assign
```
  dest = src.value_or(dest);
```

ricky122-5 committed:
refactor: Rename sfield names, sfExtensionComputeLimit -> sfGasLimit, sfExtensionSizeLimit -> sfBytecodeSizeLimit (#7708)

mvadari committed:
Merge branch 'ripple/se/wasmi-tests' of https://github.com/XRPLF/rippled into ripple/se/fees

mvadari committed:
fix build

mvadari committed:
Merge pull request #7857 from XRPLF/mvadari/se-fees-update-july

chore: Update `ripple/se/fees` branch from its base
